connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/ConnectRestServerTest.java

```
@@ -375,6 +395,44 @@ public void testDefaultCustomizedHttpResponseHeaders() throws IOException  {
         checkCustomizedHttpResponseHeaders(headerConfig, expectedHeaders);
     }
 
+    static final class MonitorableConnectRestExtension extends PluginsTest.TestConnectRestExtension implements Monitorable {
+
+        private boolean called = false;
+        private static MetricName metricName;
+
+        @Override
+        public void register(ConnectRestExtensionContext restPluginContext) {
+            called = true;
+        }
+
+        @Override
+        public void withPluginMetrics(PluginMetrics metrics) {
+            metricName = metrics.metricName("name", "description", Map.of());
+            metrics.addMetric(metricName, (Gauge<Boolean>) (config, now) -> called);
+        }
+    }
+
+    @Test
+    public void testMonitorableConnectRestExtension() {
+        Map<String, String> configMap = new HashMap<>(baseServerProps());
+        configMap.put(RestServerConfig.REST_EXTENSION_CLASSES_CONFIG, MonitorableConnectRestExtension.class.getName());
+
+        doReturn(plugins).when(herder).plugins();
+        doReturn(Collections.singletonList(new MonitorableConnectRestExtension())).when(plugins).newPlugins(any(), any(), any());
+        server = new ConnectRestServer(null, restClient, configMap);
+
+        // the call throws because of mocks but the ConnectRestExtension should have been initialized
+        assertThrows(ConnectException.class, () -> server.initializeResources(herder));
```

nit: Are some mocks missing? This looks like the test is asserting it contains a bug.
I'm suprised that strict stubs doesn't also throw after the test completes to prevent discarding errors like this.

---

I had forgotten about this hack, I've now fixed the mocks so the `initializeResources()` call does not throw.